webapp/Connector/src/view/Header.js

```diff
@@ -119,3 +119,86 @@ Ext.define("Connector.view.Header", {
         this.callParent();
     }
 });
+
+Ext.define('Connector.view.AnnouncementHeader', {
+
+    extend : 'Ext.panel.Panel',
+
+    layout : {
+        type : 'hbox',
+        align : 'middle'
+    },
+
+    alias : 'widget.announcementheader',
+
+    hidden : true,
+
+    cls: 'announcementheader',
+
+    initComponent : function() {
+
+        this.msg = this.getMessages();
+        this.items = [
+            {
+                xtype: 'box',
+                tpl: ['<img src="{imgSrc}" width="20" height="20">'],
+                data: {imgSrc: Connector.resourceContext.imgPath + '/' + 'warning_indicator.svg'},
+                margin : 15
+            },
+            this.msg,
+            {
+                xtype : 'button',
+                text : 'x',
+                margin : 15,
+                tooltip : 'dismiss this notification',
+                handler : function(){
+                    LABKEY.Ajax.request({
+                        url : LABKEY.ActionURL.buildURL("core", "dismissCoreWarnings.api"),
+                        method : 'POST',
+                        scope : this,
+                        success : function (){
+                            this.setVisible(false);
+                        },
+                        failure : function(){
+                            Ext.Msg.alert('Failed to dismiss warnings');
+                        }
+                    });
+                },
+                scope : this
+            }
+        ];
+
+        LABKEY.Ajax.request({
+            url: LABKEY.ActionURL.buildURL("cds", "getDismissableWarnings.api"),
+            method: 'GET',
+            scope : this,
+            success: function (response){
+                var o = Ext.decode(response.responseText);
+                if (o.messages) {
+                    this.setVisible(true);
+                    this.msg.update(o.messages);
+                }
+                else
+                    this.setVisible(false);
+            }
+        }, this);
+
+        this.callParent();
+    },
+
+    getMessages : function(){
+        var tpl = new Ext.XTemplate(
+            '<tpl for=".">',
+            '<div class="notification-messages">',
+            '<span>{.}</span><br>',
+            '</div>',
+            '</tpl>'
+        );
+
+        return new Ext.Component({
+            tpl : tpl,
+            flex : 2,
+            margin : '15, 5, 15, 5'
```

Comment:
I don't think I have seen this syntax for the margin prop. I would have expected it to accept a single int value for the case where all sides are the same or a string like the following: `margin: '15px 5px'` for this case.